### ISSUE TITLE
torch batchwise max with indices

### ISSUE THREAD

**Issue by sbagchi12** [pyg-team/pytorch_geometric#943]
## 🚀 Feature
A batchwise verison of torch.max() or torch.argmax().

## Motivation
I have an array with elements coming from multiple examples where each example contributes an independent number of elements. I would like to have an efficient way to find the max element along with the index.

## Pitch

torch.max() should support batches of max operations instead of a single tensor

## Alternatives

A hacky alternative 

c = torch.stack([torch.argmax(a[batch==i],dim=0,keepdim=True) for i in range(2)],dim=0)

**Comment by rusty1s** [comment]
This is exactly what `torch-scatter` is for, and we make use of it all the time, *e.g.* when aggregating neighboring node features or performing global pooling. Your example can be thus enhanced via:
```python
from torch_scatter import scatter_max
out, out_arg = scatter_max(a, batch, dim=0, dim_size=2)
```

**Comment by sbagchi12** [comment]
That works perfectly, thanks. It was not clear from the documentation. I did not see any mention of 'batch'.
